### PR TITLE
Show experience points in frontend and have it dynamically update the datastore.

### PR DESCRIPTION
**Pull request by lattai**

### CODE REVIEW

File: src/main/webapp/gameDialogue.js [hunk 1]
```diff
@@ -93,14 +98,60 @@ function modifyIfFinalStage() {
...
-function playmusic() {
-  //TODO: change the play and pause to toggle on and off icons
-  if(document.getElementById('player-button').innerText == "PAUSE"){
-    document.getElementById('player-button').innerText = "PLAY";
+function getExp() {
+  fetch("/experience")
+  .then(response => response.text())
+  .then(expString => {
+    exp = parseInt(expString);
+    console.log("EXPERIENCE: " + exp);
```
Comment (lzoebaker) [inline src/main/webapp/gameDialogue.js:106]: do we need this console.log() statement?

File: src/main/webapp/gameDialogue.js [hunk 2]
```diff
@@ -93,14 +98,60 @@ function modifyIfFinalStage() {
...
+function workPromoButtonSwitch() {
+  if (exp == threshold) {
+    // Changes button to "TRY FOR PROMOTION"
+    const promoButton = document.getElementById('promotion-button');
+    promoButton.innerText = "TRY FOR PROMOTION";
+  } else if (exp == threshold + 1) {
+    // Changes button link to promotion.html
+    showPromoButton(true, null);
+  }
+  else {
```
Comment (lzoebaker) [inline src/main/webapp/gameDialogue.js:134]: } else {

File: src/main/webapp/gameStage.html [hunk 3]
```diff
@@ -13,28 +13,22 @@
...
-
   <h2 class="step-outline">Welcome!</h2>
   <img style="float:left;" src="images/teach-all.png" alt="pixel teacher">
-
   <div class="step-wrap, step-dialogue">
       <div>
           <p id="dialogue-container"></p>
       </div>
-      <span>
+      <spam>
```
Comment (lzoebaker) [inline src/main/webapp/gameStage.html:22]: I think maybe this commit will revert the spam --> span changes?

File: src/main/webapp/gameDialogue.js [hunk 4]
```diff
@@ -93,14 +98,60 @@ function modifyIfFinalStage() {
...
+}
+
+function workPromoButtonSwitch() {
+  if (exp == threshold) {
+    // Changes button to "TRY FOR PROMOTION"
+    const promoButton = document.getElementById('promotion-button');
+    promoButton.innerText = "TRY FOR PROMOTION";
+  } else if (exp == threshold + 1) {
+    // Changes button link to promotion.html
+    showPromoButton(true, null);
```
Comment (lzoebaker) [inline src/main/webapp/gameDialogue.js:132]: Add comment explaining the null argument.

File: src/main/webapp/gameDialogue.js [hunk 5]
```diff
@@ -15,11 +15,16 @@ const FINAL_STAGE_BUTTON_VALUE = 'COMPLETE PATH';
 const FINAL_STAGE_REDIRECTION = 'CompletedPath.html';
 
 var dialogueArray;
-var i;
+var dialogueRegex;
+var exp;
```
Comment (DirtGrubDylan) [inline src/main/webapp/gameDialogue.js:19]: experience*

File: src/main/webapp/gameDialogue.js [hunk 6]
```diff
@@ -15,11 +15,16 @@ const FINAL_STAGE_BUTTON_VALUE = 'COMPLETE PATH';
...
+var threshold = 15;
+
 function loadFunctions() {
   modifyIfFinalStage();
+  workPromoButtonSwitch();
   getImage();
   getDialogue();
+  getExp();
```
Comment (DirtGrubDylan) [inline src/main/webapp/gameDialogue.js:27]: getExperience*

File: src/main/webapp/gameDialogue.js (hunk 4 above)
Comment (DirtGrubDylan) [reply src/main/webapp/gameDialogue.js:131]: +1

File: src/main/webapp/gameDialogue.js [hunk 7]
```diff
@@ -93,14 +98,58 @@ function modifyIfFinalStage() {
...
+    promoLink.removeAttribute("href");
   }
-  document.getElementById('player').muted=!document.getElementById('player').muted;
+  promoButton.onclick = onclick;
+  promoButton.innerText = getWorkButtonTask();
+}
+
+function getWorkButtonTask() {
+  var taskArray = ["WRITE SOME CODE", "ASK MANAGER FOR FEEDBACK",
+      "ATTEND A MEETING", "TEST YOUR CODE", "FIX A BUG", "GET SOME COFFEE", "DEBUG A CODE"];
```
Comment (WillHipschman) [inline src/main/webapp/gameDialogue.js:152]: debug a code? debug some code? debug code?

**Pull request merged**